src/wallet/wallet_rpc_server.cpp

```diff
@@ -2366,14 +2372,18 @@ namespace tools
       er.message = "Failed to generate wallet";
       return false;
     }
-    if (m_wallet)
-      delete m_wallet;
     m_wallet = wal.release();
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
   bool wallet_rpc_server::on_open_wallet(const wallet_rpc::COMMAND_RPC_OPEN_WALLET::request& req, wallet_rpc::COMMAND_RPC_OPEN_WALLET::response& res, epee::json_rpc::error& er)
   {
+    if (m_wallet)
```

Comment:
There is no harm leaving this method alone. Users may have code already using open_wallet to open a new wallet.